### DISCUSSION TITLE
How to trigger file delete?

### DISCUSSION THREAD

**Discussion by BruceQin2016** [openlawlibrary/pygls#459] [Q&A]
When i try to use "WORKSPACE_DID_DELETE_FILES" feature, nothing happened after i delete one file

The code is here:

import logging
from pygls.server import LanguageServer
from lsprotocol.types import (
    TEXT_DOCUMENT_COMPLETION,
    CompletionItem,
    CompletionList,
    CompletionParams,
    CompletionOptions,
    WORKSPACE_DID_DELETE_FILES,
    DeleteFilesParams
)
from pygls.workspace import TextDocument
from lsprotocol import types

@server.feature(WORKSPACE_DID_DELETE_FILES)
def did_delete(params:DeleteFilesParams):
     logging.info(params.files)

no log output
how to deal with this?

**Comment by alcarney** [comment]
As well as registering the `WORKSPACE_DID_DELETE_FILES` handler, you have to tell the client which files you are interested in seeing notifications for.

For example, the following snippet will ask the client to send you delete notifications for any python files in the workspace.

```python
@server.feature(
    WORKSPACE_DID_DELETE_FILES,
    FileOperationRegistrationOptions(
        filters=[
            FileOperationFilter(
                pattern=FileOperationPattern(glob="**/*.py"),
            )
        ]
    ),
)
def did_delete(params:DeleteFilesParams):
    logging.info(params.files)
```

The [LSP Specification](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#fileOperationRegistrationOptions) has the details on all the different options you can use with `FileOperationRegistrationOptions`

Hope that helps!

**Reply by BruceQin2016** [reply]
After i addied FileOperationRegistrationOptions, it works.
That's the difference from TextDocument section.
Thank you
